base/src/main/java/proguard/evaluation/EmptyCodeAttributeException.java

```diff
@@ -28,6 +26,6 @@ public class EmptyCodeAttributeException
     public EmptyCodeAttributeException(String message)
     {
         //ToDo: update component error id
-        super(message, 3, Collections.emptyList());
+        super(3, message);
```

Comment:
Since you added some new IDs now, can you make this `3` and `2` constants in `ErrorId`

Reply:
@rubenpieters There is still a ToDo here from you, I think. Should I change the ID to also use ranges? If yes, to what?

Reply:
@rubenpieters I think this ToDo is still from you? Should I change the ID in this PR also? If yes, to what ranges should they map?

Reply:
I took the liberty of removing the Todo and putting it in the range of the PartialEvaluator since it is in that package. + moved it to the exception subdir we made.